src/fiatconnect/ReviewScreen.tsx

```diff
@@ -34,6 +40,28 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
   const { t } = useTranslation()
   const dispatch = useDispatch()
   const { flow, normalizedQuote, fiatAccount } = route.params
+  const providerId = normalizedQuote.getProviderId()
+
+  const fiatConnectQuotesLoading = useSelector(fiatConnectQuotesLoadingSelector)
+  const fiatConnectQuotes = useSelector(fiatConnectQuotesSelector)
+  const [showingExpiredQuoteDialog, setShowingExpiredQuoteDialog] = useState(false)
+
+  const updatedQuoteData = fiatConnectQuotes.find(
+    (quote) => quote.provider.id === providerId && quote.ok
```

Comment:
What if the quote is not found? what if the quote is not ok?

Reply:
In that scenario, updatedQuoteData is undefined. The quote variable defined in line 52 is the original normalizedQuote and it will not be updated in line 54. Because the quote is still expired, the expiredQuoteDialog will still be shown and the client will need to re-obtain a quote. This will repeat until the provider's quote is found and ok and has a valid timestamp.

Reply:
I think it would be helpful to communicate to the user that there is an error. An error banner at the top saying there was an error fetching information from the provider would help.

Reply:
That's a good idea, will implement it.